packages/cli/src/executions/executions.controller.ts

```diff
@@ -110,4 +110,23 @@ export class ExecutionsController {
 
 		return await this.executionService.delete(req, workflowIds);
 	}
+
+	@Patch('/:id')
+	async update(req: ExecutionRequest.Update) {
+		if (!isPositiveInteger(req.params.id)) {
+			throw new BadRequestError('Execution ID is not a number');
+		}
+
+		const workflowIds = await this.getAccessibleWorkflowIds(req.user, 'workflow:read');
+
+		if (workflowIds.length === 0) throw new NotFoundError('Execution not found');
+
+		const { tags, vote } = req.body;
```

Comment:
We should validate these parameters. Unfortunately we are still lacking unified way to do that. One example is [here](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/controllers/me.controller.ts#L207-L213) that uses`class-validator` and `class-transformer`. Unfortunately that is not really elegant and has its own issues. I'm fine doing either simple manual typecheck here or using the `class-validator` approach until we have a proper way to do validation

Reply:
I made a simple validation using zod, is that ok?